xarray/core/duck_array_ops.py

```diff
@@ -531,7 +531,10 @@ def pd_timedelta_to_float(value, datetime_unit):
 
 
 def _timedelta_to_seconds(array):
-    return np.reshape([a.total_seconds() for a in array.ravel()], array.shape) * 1e6
+    if isinstance(array, datetime.timedelta):
+        return array.total_seconds() * 1e6
+    else:
+        return np.reshape([a.total_seconds() for a in array.ravel()], array.shape) * 1e6
```

Comment:
This ravel + reshape is potentially [quite bad](https://docs.dask.org/en/stable/array-chunks.html#reshaping) for nD chunked arrays.

I think we want the equivalent of `apply_ufunc(..., vectorize=True, dask="parallelized")`. I was going to suggest `np.vectorize` but there is no `dask.array.vectorize` AFAICT. We could also just merge with a comment and raise an issue to fix later.

Reply:
I think this is applied strictly on NumPy arrays through `map_blocks`, though, so the performance of dask's reshape should not be a problem, right?

Reply:
Oh right 🤦🏾 I see that below. Looks good to me!